### PR TITLE
fix: remove requests that error from queue

### PR DESCRIPTION
**Pull request by jmannau**
fixes https://github.com/ngneat/cashew/issues/9

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~ (n/a)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

This PR updates the library to remove requests from the cache queue if the request fails.

Issue Number: https://github.com/ngneat/cashew/issues/9

## What is the new behavior?

Any request that errors is removed from the cache queue. This prevents stale requests being sent from the queue if a request fails.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngneat/cashew/src/lib/httpCacheInterceptor.ts [hunk 1]
```diff
@@ -34,10 +34,15 @@ export class HttpCacheInterceptor implements HttpInterceptor {
...
-          if (event instanceof HttpResponse) {
-            const cache = this.httpCacheManager._resolveResponse(event);
-            this.httpCacheManager._set(key, cache, +ttl);
+        tap({
+          next: event => {
+            if (event instanceof HttpResponse) {
+              const cache = this.httpCacheManager._resolveResponse(event);
+              this.httpCacheManager._set(key, cache, +ttl);
+            }
+            queue.delete(key);
```
Comment (NetanelBasal) [inline projects/ngneat/cashew/src/lib/httpCacheInterceptor.ts:43]: Why not on complete?

**Comment by jmannau** [comment]
Well it turns out complete is only called when the observable completes. Not when it errors as well. We could use the finalize operator which is called when the observable completes or errors.

**Comment by NetanelBasal** [comment]
Use finalize

**Comment by jmannau** [comment]
I have updated the PR to use finalize rather than next & error handlers. Thanks for your feedback.

**Pull request merged**